src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -5321,6 +5316,12 @@ NullableContextOptions getDefaultState()
                 _ when getDefaultState().WarningsEnabled() => NullableContext.WarningsContextInherited | NullableContext.WarningsEnabled,
                 _ => NullableContext.WarningsContextInherited,
             };
+
+            // IsGeneratedCode might be slow, only call it when needed:
+            NullableContextOptions getDefaultState()
+                => lazyDefaultState ??= syntaxTree.IsGeneratedCode(Compilation.Options.SyntaxTreeOptionsProvider, CancellationToken.None)
```

Comment:
And all the work the LDM made me put into making this evaluate a `bool` instead of a `bool?` pays off 😆.